libmscore/scorefile.cpp

```diff
@@ -428,29 +457,15 @@ bool MasterScore::saveFile()
             // rename old file into backup
             //
             if (dir.exists(basename)) {
-                  if (!dir.rename(basename, backupName)) {
+                  if (!QFile::rename(name, backupDirString + QString(QDir::separator()) + backupName)) {
```

Comment:
Quess we'd wind out here whether the directory was wrtitable.
Only we're telling nobdy that it failed...

Reply:
Seems a case of "never check for error conditions you can't/don't handle"

Reply:
I'm sorry but are you just exclaiming or stating that some changes are needed?

Reply:
Not your fault, you just inherited that boghus behavoir.
Well, 2 options, as I (tought to have) implied:

1. just use that `QFile::rename()` and don't check whether it worked, we're not doing anything with that information anyway.
2. Actually do emit an error message if the rename fails. Not possible here, not via a dialog at least, as we're in libmscore, but at leate a `QDebug()` or `QError()` should be done here.

Reply:
Actually QtCriticalMsg or QtWarningMsg, not QtFatalMsg

Reply:
When and where are those QWarning()/QError()/QAssert() displayed? When debugging?

Reply:
Or when starting MuseScore from the commandline (at least on non-Windows, there extra steps are needed)

Reply:
See https://github.com/musescore/MuseScore/pull/5380#discussion_r334229114.